tsl/src/bgw_policy/job.c

```diff
@@ -55,6 +55,29 @@
 
 #define REORDER_SKIP_RECENT_DIM_SLICES_N 3
 
+static void
+log_retention_boundary(int elevel, PolicyRetentionData *policy_data, const char *message)
+{
+	char *relname;
+	Datum boundary;
+	Oid outfuncid = InvalidOid;
+	bool isvarlena;
+
+	getTypeOutputInfo(policy_data->boundary_type, &outfuncid, &isvarlena);
+
+	relname = get_rel_name(policy_data->object_relid);
+	boundary = policy_data->boundary;
+
+	if (outfuncid != InvalidOid)
+		elog(elevel,
+			 "%s \"%s\": dropping data older than %s",
+			 message,
+			 relname,
+			 DatumGetCString(OidFunctionCall1(outfuncid, boundary)));
+	else
+		ereport(ERROR, errcode(ERRCODE_INVALID_PARAMETER_VALUE), errmsg("Invalid boundary type"));
```

Comment:
If this function is eventually conditionally called then we can't be throwing exceptions here depending on the `verbose_log` level. An error logging function should never influence the result of execution.